### PR TITLE
Set the hostname to the container ID

### PR DESCRIPTION
**Pull request by caglar10ur**
And welcome to a world where we require dbus to do that /s

### CODE REVIEW

File: isos/bootstrap-staging.sh [hunk 1]
```diff
@@ -106,6 +106,7 @@ yum_cached -c $cache -u -p $PKGDIR install \
     libtirpc \
     grep \
     haveged \
+    dbus \
```
Comment (fdawg4l) [inline isos/bootstrap-staging.sh]: Curious.  We can't just echo hostname > /etc/hostname  and set /etc/hosts appropriately?

Comment (caglar10ur) [inline isos/bootstrap-staging.sh]: That won't change the hostname unless you reboot one more time

Comment (caglar10ur) [inline isos/bootstrap-staging.sh]: By the way the other alternative might be calling the syscall directly from the tether to set the hostname.

But in every case not sure why systemd photon package does not depend on dbus (or whether it's OK to not to have it). It looks like dbus is a hard requirement for systemd to work.

**Comment by dougm** [comment]
lgtm

**Comment by gigawhitlocks** [comment]
:sheep: :it:

**Comment by hickeng** [comment]
@caglar10ur did the SetHostname function in cmd/tether/utils/hostname_linux.go not work?

File: isos/bootstrap/bootstrap [hunk 2]
```diff
@@ -51,11 +51,11 @@ if mount -t ext4 /dev/disk/by-label/containerfs ${MOUNTPOINT}; then
     #	rm -f /mnt/.tether/volumes/containerfs
 
     # set up the hostname
-    #	if [ -f /mnt/etc/hostname ]; then
-    #		/bin/hostname -F /mnt/etc/hostname
-    #	else
-    #		/bin/hostname -F /mnt/.tether-init/hostname
-    #	fi
+    if [ -f ${MOUNTPOINT}/etc/hostname ]; then
```
Comment (hmahmood) [inline isos/bootstrap/bootstrap]: Don't these two files point to the same location?

**Comment by caglar10ur** [comment]
@hickeng as far as I see no one is calling it and setting t in init script is how we did in bonneville. We could close this and start setting hostname via tether as I said above (https://github.com/vmware/vic/pull/422#discussion_r58583066)

File: isos/bootstrap/bootstrap (hunk 2 above)
Comment (caglar10ur) [inline isos/bootstrap/bootstrap]: ` 
# if we don't have a populated init layer, pull from guestinfo
28     if [ ! -f ${MOUNTPOINT}/.tether-init/docker-id ]; then
...
 48     fi
`

Comment (hmahmood) [inline isos/bootstrap/bootstrap]: Does the if on line 54 imply we may not have /etc/hostname file? If yes, would that be a problem?

**Comment by hickeng** [comment]
@caglar10ur prefer to do it in the tether - it needs to end up there eventually

**Comment by caglar10ur** [comment]
Closing this. Tether will set the hostname

**Pull request closed without merging**